Initialize gem

mkostick opened the pull request:
There are two commits here.  The first is the result of running ore:
`mine yeti_exception --travis --yard --version=1.0.0 --author="Yesware, Inc." --email="engineering@yesware.com"`

I intentionally did not use the Yesware ore template, since it adds internal things that won't be useful here.

The second commit includes edits to align with Yesware conventions and development ease.

Commit:
Result of ore creation

Commit:
Updates based on Yesware defaults

mkostick commented:
Note to reviewers:

This gem is meant to work with https://github.com/Yesware/yeti_logger, which is also a public repo, so it mirrors its use of Travis CI and the MIT license.

mkostick commented:
I need to look into the CI errors.

Also, just for direction:  This gem is going to be an encapsulation of an exception pattern we started using on a new app recently.

mike-yesware commented on lib/yeti_exception/version.rb:
```diff
@@ -0,0 +1,4 @@
+module YetiException
+  # yeti_exception version
+  VERSION = "1.0.0"
```
[nit] should this be lower or `pre` until you're ready for a real release version?

mkostick replied:
Sure, I can do that until I'm ready to publish

mike-yesware reviewed (commented):
One comment

Commit:
Review feedback and CI fix

- Tag version with `-pre` until ready for release
- Travis CI apparently fails if summary/description contains TODO

meghan-mcqueeney commented on ChangeLog.md:
```diff
@@ -0,0 +1,4 @@
+### 1.0.0 / 2020-04-29
```
do you want to update this documentation to match the "pre"?

meghan-mcqueeney commented on yeti_exception.gemspec:
```diff
@@ -0,0 +1,37 @@
+  gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
+  gem.require_paths = ['lib']
+
+  gem.add_development_dependency 'bundler', '~> 1.10'
```
do we want restrictions like these?

Commit:
Update to recent gem versions

meghan-mcqueeney commented:
G2G once CI passes.

Commit:
Review and CI updates

- Empty changelog until ready for actual release
- install bundler manually so we're ensured v2.0

mkostick commented:
All review comments addressed, and CI is finally green
